ssg/products.py

```diff
@@ -4,17 +4,70 @@
 import os
 from collections import namedtuple
 
-from .constants import product_directories
+from .build_cpe import ProductCPEs
+from .constants import (product_directories,
+                        DEFAULT_UID_MIN,
+                        PKG_MANAGER_TO_SYSTEM,
+                        PKG_MANAGER_TO_CONFIG_FILE,
+                        XCCDF_PLATFORM_TO_PACKAGE)
+from .utils import merge_dicts
 from .yaml import open_raw
 
 
-def get_product_yaml(ssg_root, product):
+def _validate_product_oval_feed_url(contents):
+    if "oval_feed_url" not in contents:
+        return
+    url = contents["oval_feed_url"]
+    if not url.startswith("https"):
+        msg = (
+            "OVAL feed of product '{product}' is not available through an encrypted channel: {url}"
+            .format(product=contents["product"], url=url)
+        )
+        raise ValueError(msg)
+
+
+def _get_implied_properties(existing_properties):
+    result = dict()
+    if "pkg_manager" in existing_properties:
+        pkg_manager = existing_properties["pkg_manager"]
+        if "pkg_system" not in existing_properties:
+            result["pkg_system"] = PKG_MANAGER_TO_SYSTEM[pkg_manager]
+        if "pkg_manager_config_file" not in existing_properties:
+            if pkg_manager in PKG_MANAGER_TO_CONFIG_FILE:
+                result["pkg_manager_config_file"] = PKG_MANAGER_TO_CONFIG_FILE[pkg_manager]
+
+    if "uid_min" not in existing_properties:
+        result["uid_min"] = DEFAULT_UID_MIN
+
+    if "auid" not in existing_properties:
+        result["auid"] = existing_properties.get("uid_min", DEFAULT_UID_MIN)
+
+    return result
+
+
+def get_product_yaml(product_yaml_path):
```

Comment:
Two questions here:

1. Does this need to be tempered by the build YAML? If you look at e.g., `combine_ovals.py`:
    ```python3
    env_yaml = ssg.yaml.open_environment(args.build_config_yaml, args.product_yaml)
    ```
2. Would something like `load_product_yaml` be clearer? Especially if we add a `get_product_yaml_path(product)` -- we could make the product YAML path be inferred by the actual product. Just a thought.

Reply:
I like the idea of the two functions.

Reply:
I just renamed the current function to `load_product_yaml`.